wasabi/printer.py

```diff
@@ -67,67 +65,127 @@ def __init__(
+        """
         return self._get_msg(
             title, text, style=MESSAGES.WARN, show=show, spaced=spaced, exits=exits
         )
 
-    def info(self, title="", text="", show=True, spaced=False, exits=None):
-        """Print an informational message."""
+    def info(
+        self,
+        title: str = "",
+        text: str = "",
+        show: bool = True,
+        spaced: bool = False,
+        exits: Optional[int] = None,
+    ):
+        """Print an informational message.
+
+        title (str): The main text to print.
+        text (str): Optional additional text to print.
+        show (bool): Whether to print or not. Can be used to only output
+            messages under certain condition, e.g. if --verbose flag is set.
+        spaced (bool): Whether to add newlines around the output.
+        exits (int): Optional toggle to perform a system exit.
+        """
         return self._get_msg(
             title, text, style=MESSAGES.INFO, show=show, spaced=spaced, exits=exits
         )
 
     def text(
         self,
-        title="",
-        text="",
-        color=None,
-        bg_color=None,
-        icon=None,
-        spaced=False,
-        show=True,
-        no_print=False,
-        exits=None,
+        title: str = "",
+        text: str = "",
+        color: Optional[Union[str, int]] = None,
+        bg_color: Optional[Union[str, int]] = None,
+        icon: Optional[str] = None,
+        spaced: bool = False,
+        show: bool = True,
+        no_print: bool = False,
+        exits: Optional[int] = None,
     ):
         """Print a message.
 
-        title (unicode): The main text to print.
-        text (unicode): Optional additional text to print.
-        color (unicode / int): Foreground color.
-        bg_color (unicode / int): Background color.
-        icon (unicode): Name of icon to add.
-        spaced (unicode): Whether to add newlines around the output.
+        title (str): The main text to print.
+        text (str): Optional additional text to print.
+        color (str / int): Optional foreground color.
+        bg_color (str / int): Optional background color.
+        icon (str): Optional name of icon to add.
+        spaced (bool): Whether to add newlines around the output.
         show (bool): Whether to print or not. Can be used to only output
             messages under certain condition, e.g. if --verbose flag is set.
         no_print (bool): Don't actually print, just return.
-        exits (int): Perform a system exit.
+        exits (int): Perform a system exit. Optional.
         """
         if not show:
             return
         if self.pretty:
-            color = self.colors.get(color, color)
-            bg_color = self.colors.get(bg_color, bg_color)
-            icon = self.icons.get(icon)
+            color = self.colors.get(cast(str, color), color)
+            bg_color = self.colors.get(cast(str, bg_color), bg_color)
+            icon = self.icons.get(cast(str, icon))
```

Comment:
There are some value re-assignment happening here. So I opted to just type-cast the variable rather than creating a new one.